### ISSUE TITLE
SQLlite crash - locked - code 5

### ISSUE THREAD

**Issue by anthonyhumphreys** [mauron85/react-native-background-geolocation#312]
## Your Environment

* Plugin version: ^0.5.0-alpha.44
* Platform: Android
* OS version: 7
* Device manufacturer and model: Blackview A7
* React Native version: 0.57.3
* Plugin configuration options:
```
BackgroundGeolocation.configure({
      desiredAccuracy: BackgroundGeolocation.HIGH_ACCURACY,
      stationaryRadius: 50,
      distanceFilter: 50,
      notificationTitle: 'Background tracking',
      notificationText: 'enabled',
      debug: false,
      startOnBoot: false,
      stopOnTerminate: false,
      locationProvider: BackgroundGeolocation.ACTIVITY_PROVIDER,
      interval: 20000,
      fastestInterval: 20000,
      activitiesInterval: 20000,
      stopOnStillActivity: false,
      url: 'https://api:8001/api/v1/Tracking/activator/send',
      syncUrl: 'https://api:8001/api/v1/Tracking/activator/sync',
      httpHeaders: { Authorization: `Bearer ${this.props.token}` },
      postTemplate: {
        sessionId: this.props.sessionId,
        userId: this.props.userID,
        latitude: '@latitude',
        longitude: '@longitude',
        dateRegistered: moment().utc().format(),
        isSecurityWatching: this.props.securityAdded,
        accuracy: 0,
      },
    });
```
* Link to your project:

## Context

Android is crashing out while location enabled - getting the attached log in our exception collector.
## Expected Behavior

...not crash :p
## Actual Behavior

After a few seconds the app crashes with the error in the log attached - sqllite database locked code 5
## Steps to Reproduce

Using configuration given, call BackgroundGeolocation.start();
## Debug logs

[app-f7493081-1177-4ee0-b97b-9b9c28c0ed6d-id-9de74193-dad9-463c-8d70-b8c73ce653da.txt](https://github.com/mauron85/react-native-background-geolocation/files/2525083/app-f7493081-1177-4ee0-b97b-9b9c28c0ed6d-id-9de74193-dad9-463c-8d70-b8c73ce653da.txt)

**Comment by mauron85** [comment]
I see you're using 'stopOnTerminate: false'. It's those exact steps how to reproduce the issue, or is there also killing app and reopening it later involved?

**Comment by anthonyhumphreys** [comment]
Huh - that's interesting, i hadn't spotted that. I have a suspicion multiple readers are being spawned? Stab in the dark guess.

The steps are exact, basically we have a button that triggers BackgroundGeolocation.start() -= leaving it on that screen it just crashes after a short period of time.

This doesn't happen on better devices though it seems. Could just be that its a dodgy handset with limited resources

**Comment by mauron85** [comment]
Ok. If you don't need `stopOnTerminate: false` rather don't use it. It brings some problems on some devices. I've changed documentation to use true value instead. 

But if your steps are exact, then it's very interesting as android sqlite should be thread safe and all write operation should wait for previous one to complete, rather than throw db locked exception, when single SQLiteHelper is used (which should be case for this plugin).

Maybe sqlite was compiled with some unusual compiler flags for you device. If I find something I'll post an update.

**Comment by anthonyhumphreys** [comment]
Awesome, thank you!

**Comment by mauron85** [comment]
However there might be one edge case, when service is done with location syncing - accessing database to mark them all as successfully synced and in if same time new location arrive, then db might be locked by sync thread. However this should happen very rarely, definitely not every time you start the app (and also hard to fix).

**Comment by mauron85** [comment]
Is this still an issue?

**Comment by mauron85** [comment]
FYI just released version 0.5.alpha.47. The edge case mentioned should be fixed. Please give it a try and report.

**Comment by mauron85** [comment]
Closing as resolved. Feel free to reopen when needed.
